swayfmt/src/items/item_abi.rs

```diff
@@ -21,20 +22,54 @@ impl Format for ItemAbi {
         self.name.format(formatted_code, formatter)?;
         Self::open_curly_brace(formatted_code, formatter)?;
 
+        let abi_items = self.abi_items.get();
+
+        // add pre fn_signature comments
+        let end = if abi_items.is_empty() {
+            self.span().end()
+        }
+        // if there are existing abi items, we want to end before the hash token
+        // This is OK to unwrap since we've checked is_empty() above.
+        else if let Some(first_attr) = abi_items.first().unwrap().0.attribute_list.first() {
+            first_attr.hash_token.span().start()
+        } else {
+            abi_items.first().unwrap().0.value.span().start()
+        };
+
+        write_comments(
+            formatted_code,
+            std::ops::Range {
+                start: self.name.span().end(),
+                end,
+            },
```

Comment:
I'm starting to think using `std::ops::Range` as an input doesn't look as nice, because in most cases it's formatted into this and takes up a lot of vertical space. Is it worth considering going back to just providing `usize, usize`?